pkg/extractor/extractor.go

```diff
@@ -76,29 +84,72 @@ func ParseYaml(content string) (*[]Configuration, error) {
 	}
 }
 
+func ParseJsonToK8sValues(content []byte) (map[string]interface{}, error) {
+	var values map[string]interface{}
+	err := json.Unmarshal(content, &values)
+	if err != nil {
+		return nil, err
+	}
+	return values, nil
+}
+
 func extractYamlConfigurations(content string) (*[]Configuration, error) {
 	var configurations []Configuration
 
 	yamlDecoder := yaml.NewDecoder(bytes.NewReader([]byte(content)))
 
 	var err error
 	for {
-		var doc = map[string]interface{}{}
-		err = yamlDecoder.Decode(&doc)
+		var anyTypeStore interface{}
+		err = yamlDecoder.Decode(&anyTypeStore)
 		if err != nil {
-			break
+			if err == io.EOF {
+				break
+			} else {
+				return nil, err
+			}
 		}
 
-		if len(doc) > 0 {
-			configurations = append(configurations, doc)
+		yamlByteArray, err := yaml.Marshal(anyTypeStore)
+		if err != nil {
+			return nil, err
+		}
+
+		jsonByte, err := yamlConvertor.YAMLToJSON(yamlByteArray)
+		if err != nil {
+			return nil, err
+		}
+
+		configurations = append(configurations, extractMetadataNameAndKind(jsonByte))
+	}
+
+	return &configurations, nil
+}
+
+func extractMetadataNameAndKind(content []byte) Configuration {
```

Comment:
I think we should rename to something like `extractConfigurationMetadata` since it get's now also Annotations, apiVersion etc.